colossalai/inference/core/engine.py

```diff
@@ -42,19 +43,26 @@ class InferenceEngine:
     def __init__(
         self,
         model: nn.Module,
-        tokenizer: Union[PreTrainedTokenizer, PreTrainedTokenizerFast],
-        inference_config: Optional["InferenceConfig"] = None,
+        inference_config: InferenceConfig,
+        tokenizer: Optional[Union[PreTrainedTokenizer, PreTrainedTokenizerFast]] = None,
         verbose: bool = False,
         model_policy: Policy = None,
     ) -> None:
         assert inference_config, "Please provide inference_config."
-        self.tokenizer = tokenizer
-        self.tokenizer.pad_token = self.tokenizer.eos_token
         self.inference_config = inference_config
         self.model_config = model.config
         self.device = torch.device("cuda")
         self.dtype = inference_config.dtype
-
+        if tokenizer is None:
+            tokenizer = get_tokenizer(
+                inference_config.tokenizer,
+                trust_remote_code=inference_config.trust_remote_code,
+                tokenizer_revision=inference_config.tokenizer_revision,
+                revision=inference_config.revision,
+            )
```

Comment:
You can always just ask the user to pass in the tokenizer for simplicity.

Reply:
maybe we can also allow type `str`

Reply:
fixed

Reply:
del the tokenizer logic